### PR TITLE
feat(feature-toggle): resolve feature flags on component level

### PR DESCRIPTION
**Pull request by divyanshiGupta**
Parent Story: https://openshift.io/openshiftio/Openshift_io/plan/detail/1110

This PR adds resolving feature flags at component level
- Adds a feature-wrapper ie `f8-feature-toggle` component on top of the feature, if `showFeatureOptIn` is `true` then if the feature is not enabled default warning page with a feature opt-in button will be displayed and after opting in the feature will be displayed.

Note : 
Build is failing because it has a dependency on ngx-feature-flag's new code but the PR is yet to get merged.
Merge this PR only after this PR https://github.com/fabric8-ui/ngx-feature-flag/pull/22 is merged and ngx-feature-flag is moved to monorepo.
feature-toggle

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @divyanshiGupta for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

File: packages/planner/src/app/components_ngrx/planner-board/planner-board.component.html [hunk 1]
```diff
@@ -1,71 +1,83 @@
...
-      (sidePanelStateChange)="togglePanelState($event)"
-      [sidePanelContent]="sidePanel"
-      [sectionContent]="sectionPanel"
-    >
-    </alm-planner-layout>
-    <work-item-preview-panel #quickPreview [context]="'board'"> </work-item-preview-panel>
-  </main>
-</div>
+<f8-feature-toggle
+  featureName="Planner"
```
Comment (rohitkrai03) [inline packages/planner/src/app/components_ngrx/planner-board/planner-board.component.html]: @divyanshiGupta Should board view be on the same feature level as planner? Planner is on beta, i think board should be on same level as Query tab which is internal. @nimishamukherjee What do you think?

File: packages/fabric8-ui/src/app/app.module.ts [hunk 2]
```diff
@@ -184,6 +184,7 @@ export type StoreType = {
     ErrorService,
     FeatureFlagResolver,
     FeatureTogglesService,
+    EnableFeatureService,
```
Comment (rohitkrai03) [inline packages/fabric8-ui/src/app/app.module.ts]: Providers should be sorted alphabetically.

File: packages/planner/src/app/components_ngrx/planner-board/planner-board.component.html (hunk 1 above)
Comment (divyanshiGupta) [reply packages/planner/src/app/components_ngrx/planner-board/planner-board.component.html]: Yes, I think it should be on the same level as query tab. I will change that.

**Comment by joshuawilson** [comment]
please rebase

**Pull request merged**